x-pack/plugins/reporting/server/export_types/csv/create_job.ts

```diff
@@ -21,10 +21,12 @@ export const createJobFnFactory: CreateJobFnFactory<
     );
 
     const savedObjectsClient = context.core.savedObjects.client;
```

Comment:
actually, we shouldnt be using saved objects client here. we should use data.indexPatterns service. can you please update that ?

Reply:
The problem is that the `context.core` is a [setup](https://github.com/elastic/kibana/blob/master/x-pack/plugins/reporting/server/plugin.ts#L46:L46) contract that doesn't have the data plugin instance. I can work that around, but this is a deprecated job type that should be removed in 8.0. And the issue is labeled as `v8.0.0` only. Fixing that here will produce extra work with removing that later without any impact on the final result.